docs/features/source-generators.md

```diff
@@ -0,0 +1,285 @@
+# Source Generators
+
+
```

Comment:
❔ Do we have a way to indicate at the start that this is just a proposal, and not actually a feature that is already implemented?

Reply:
It's in the feature branch, I think that implies it?

Reply:
When @sharwell initially made this comment the PR was targeting master. @chsienki then changed to target the feature branch, partially in response to this. That's why the comment seems confusing.

Reply:
At this point though it should be clear this isn't a shipped feature though given the entirety of the documentation is in a feature branch 😄